### PR TITLE
Add presubmit prow job to run unit tests on s390x cluster

### PR DESCRIPTION
**Pull request by vamsikrishna-siddu**
Hi @brianmcarey 

I am adding a periodic and pre submit job for running unit tests on s390x.
Please do review when you have some time.

### CODE REVIEW

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml
Comment (brianmcarey) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml:812]:
```suggestion
      image: quay.io/kubevirtci/golang:v20240306-cfcbc5c
```

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (brianmcarey) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:441]:
```suggestion
        image: quay.io/kubevirtci/golang:v20240306-cfcbc5c
```

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (brianmcarey) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:433]: Can you add `skip_report: true` as initially we don't want this reporting on PRs until we know it is stable?

**Review by brianmcarey — COMMENTED** [review]
Thanks @vamsikrishna-siddu - looks good - can you just update the imge to the bootstrap image that includes go installed?

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml
Comment (vamsikrishna-siddu) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml]: done

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (vamsikrishna-siddu) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:433]: done

**Comment by vamsikrishna-siddu** [comment]
> Thanks @vamsikrishna-siddu - looks good - can you just update the imge to the bootstrap image that includes go installed?

Hi @brianmcarey  I have addressed your review comments. Can you please review it again. Thanks

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (brianmcarey) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:429]: This should be set to true as we don't have a builder image available for s390x yet. 

I was able to run the unit tests - with unnested set to true.

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml
Comment (brianmcarey) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml:799]: This should be set to true as we don't have a builder image available for s390x yet. 

I was able to run the unit tests - with unnested set to true.

**Comment by vamsikrishna-siddu** [comment]
Hi @brianmcarey I have addressed your review comments. Can you please review it again. Thanks

**Comment by brianmcarey** [comment]
Thanks @vamsikrishna-siddu  - I was able to run this against the PR enabling s390x support (https://github.com/kubevirt/kubevirt/pull/10490) - here is the link to scheduled prow job. https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/10490/pull-kubevirt-unit-test-s390x/1767539290826346496

**Review by brianmcarey — COMMENTED** [review]
@vamsikrishna-siddu - can you split the periodic job out from this? We can only add the periodic once the enablement PR is merged in kubevirt main. 

But we should be able to merge the optional presubmit lane now.

**Review by brianmcarey — COMMENTED** [review]
Just one little nit change left - could you squash the commits into one - i think this could all be handled in one commit.

**Comment by vamsikrishna-siddu** [comment]
> Just one little nit change left - could you squash the commits into one - i think this could all be handled in one commit.

Hi @brianmcarey  done

**Comment by brianmcarey** [comment]
@vamsikrishna-siddu I ran this job against the enablement PR and had some failures - https://storage.googleapis.com/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/10490/pull-kubevirt-unit-test-s390x/1800138334815653888/build-log.txt

**Comment by jschintag** [comment]
> @vamsikrishna-siddu I ran this job against the enablement PR and had some failures - https://storage.googleapis.com/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/10490/pull-kubevirt-unit-test-s390x/1800138334815653888/build-log.txt

@brianmcarey I have tested it locally and i'm relatively sure that the failures are due to the race detection that is enabled for `go-test` and only enabled by a deprecated, ineffective flag in `bazel-test`. I'm running the full go-test right now without `-race` flag to confirm, but that should be the only difference between the 2 methods.
I will also check if i can reproduce it on x86, which i think is likely the case.

**Comment by brianmcarey** [comment]
@vamsikrishna-siddu  - looks like this job is hitting a new issue?  https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_project-infra/3298/rehearsal-pull-kubevirt-unit-test-s390x/1803408968303775744

**Comment by jschintag** [comment]
> @vamsikrishna-siddu - looks like this job is hitting a new issue? https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_project-infra/3298/rehearsal-pull-kubevirt-unit-test-s390x/1803408968303775744

Interesting, i mean usbredir is not supported on s390x, and the code not included in virtctl, so if there is a new test, the failure does not surprise me.

**Comment by jschintag** [comment]
Although https://github.com/kubevirt/kubevirt/pull/12222 has been merged, it seems that the current main has some new failing unit-tests on s390x. I'm currently investigating.

**Comment by vamsikrishna-siddu** [comment]
Hi @brianmcarey  and @jschintag  I have raised this PR which fixes the unit test failures. Please review it when you get some time.
https://github.com/kubevirt/kubevirt/pull/12287

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (brianmcarey) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:433]: @vamsikrishna-siddu - since we can see that this is passing now on the periodic lane[1] - I think we can remove this `skip_report: true`

Comment (vamsikrishna-siddu) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:433]: sure @brianmcarey

Comment (vamsikrishna-siddu) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml]: Hi @brianmcarey  I have removed the skip_report: true from the PR. Can you please review it again. Thanks

**Comment by jschintag** [comment]
Looks like one new failure related to a timeout, let me take a look:
```
  [FAILED] in [It] - /home/prow/go/src/github.com/kubevirt/kubevirt/pkg/virt-handler/device-manager/device_controller_test.go:210 @ 07/10/24 07:01:32.217
  << Timeline
  [FAILED] Timed out after 5.000s.
  Expected
      <bool>: true
  to be false
  In [It] at: /home/prow/go/src/github.com/kubevirt/kubevirt/pkg/virt-handler/device-manager/device_controller_test.go:210 @ 07/10/24 07:01:32.217
------------------------------
•••••••••
Summarizing 1 Failure:
  [FAIL] Device Controller Multiple Plugins [It] should remove all device plugins if permittedHostDevices is removed from the CR
  /home/prow/go/src/github.com/kubevirt/kubevirt/pkg/virt-handler/device-manager/device_controller_test.go:210
Ran 30 of 33 Specs in 10.820 seconds
FAIL! -- 29 Passed | 1 Failed | 0 Pending | 3 Skipped
--- FAIL: TestDeviceManager (10.83s)
FAIL
FAIL	kubevirt.io/kubevirt/pkg/virt-handler/device-manager	10.971s
```

**Comment by brianmcarey** [comment]
> Looks like one new failure related to a timeout, let me take a look:
> 
> ```
>   [FAILED] in [It] - /home/prow/go/src/github.com/kubevirt/kubevirt/pkg/virt-handler/device-manager/device_controller_test.go:210 @ 07/10/24 07:01:32.217
>   << Timeline
>   [FAILED] Timed out after 5.000s.
>   Expected
>       <bool>: true
>   to be false
>   In [It] at: /home/prow/go/src/github.com/kubevirt/kubevirt/pkg/virt-handler/device-manager/device_controller_test.go:210 @ 07/10/24 07:01:32.217
> ------------------------------
> •••••••••
> Summarizing 1 Failure:
>   [FAIL] Device Controller Multiple Plugins [It] should remove all device plugins if permittedHostDevices is removed from the CR
>   /home/prow/go/src/github.com/kubevirt/kubevirt/pkg/virt-handler/device-manager/device_controller_test.go:210
> Ran 30 of 33 Specs in 10.820 seconds
> FAIL! -- 29 Passed | 1 Failed | 0 Pending | 3 Skipped
> --- FAIL: TestDeviceManager (10.83s)
> FAIL
> FAIL	kubevirt.io/kubevirt/pkg/virt-handler/device-manager	10.971s
> ```

It seems to be a flake. The last run passed - https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_project-infra/3298/rehearsal-pull-kubevirt-unit-test-s390x/1810935428258205696

**Comment by jschintag** [comment]
> It seems to be a flake. The last run passed - https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_project-infra/3298/rehearsal-pull-kubevirt-unit-test-s390x/1810935428258205696

Yes, i thought it might be, given that it was a timeout. I'll still run it a few times locally to check. Given that the timeout mentioned 5s i think it is defined in the test itself, so it should not be to hard to increase if necessary.

**Comment by vamsikrishna-siddu** [comment]
Hi @jschintag @brianmcarey the latest unit test run got passed.
https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_project-infra/3298/rehearsal-pull-kubevirt-unit-test-s390x/1810967836982513664

**Review by brianmcarey — APPROVED** [review]
The lane is optional so we should be able to take this in. It can be moved to required when it is proven to be stable. 

Thanks @vamsikrishna-siddu 

/approve

/hold cancel

**Pull request merged**
